### DISCUSSION TITLE
Creating relationships when uploading csv to Budibase DB

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#8396] [Help] [answered]
Hi all - does anyone know if it's possible to create relationships between tables when uploading a CSV to Budibase DB? I've attempted to do this but have had no luck. 

In my csv file, I've created a column matching the relationship column name. I've tried both the ID and the 'Primary search index' as the value of the related item, but neither seems to be recognised (see image below).
<img width="471" alt="image" src="https://user-images.githubusercontent.com/64409787/197870987-38cb9e89-4563-4e3a-80f3-90274b757d87.png">

Thanks!

**Comment by melohagan** [comment, marked as answer]
Hey @adamjdavies 

There is no way to do this. The tricky part here is that relationship values in the internal DB are simply GUID links to existing rows, which you would need to know ahead of time before doing your import. I can't see it working well at all.

In general there have been internal discussions around improving how relationships are handled within the Budibase DB, but nothing is set in stone.

**Reply by ghost** [reply]
Hi @melohagan 

Thanks for the response. That makes sense. I can appreciate it's not a simple task! I've actually changed the way I will be importing data, so this isn't a problem for me now. 

Thanks!

**Reply by pmankow** [reply]
@melohagan 

I think this should be revisited and added as a feature. The reason is that you are already implementing this functionality manually. When I search in a relationship cell for a value, it gives a list of similar values or an exact value if there is an exact match. In doing this you are opening the data to user error, which is understood. There would be no difference if you had a wizard do this, it would just save the user several copy/paste/clicks... If there is no match or a duplicate for certain values, those exceptions could be manually resolved by the user. 

An example of this is in nearly every BOM (bill of material) handling software out there. From Digikey, to stand-alone software packages. This would greatly increase the usefulness of your software as many people deal with BOMs and need an intermediary between using Excel and Enterprise PLM systems. You fit that bill if you could be a little more flexible on .csv importing.

Thank you.

**Comment by melohagan** [comment]
Hey @pmankow 

I actually ran into this same problem myself, and I was able to find a solution using automations.

Take a look at this page, and download the app export at the bottom: https://docs.budibase.com/docs/cascading-dropdown-filters

Then go to the **Automations** tab, and you will see how I generated relationship values for the country and city tables.

1. Create a relationship field for your imported tables
2. Run the automations. The response will timeout, however it will continue to populate the relationships in the background, which you'll see in the Data section in real time.

There is a filter in **Step 1** you will want to clear (I was trying to batch the updates, but it didn't matter).

You'll also need to update the tables / filters / JavaScript bindings to use the correct ID fields for your imported tables.

**Reply by pmankow** [reply]
Thank you, I will take a look at your solution next time I have to import a large set of data.

**Reply by JeroenP1974** [reply]
@melohagan I just imported 2000+ records and now I must manually select all the relationship data. Now I saw you solution. Downloaded it en just test it. But I got errors....

<img width="760" alt="Scherm­afbeelding 2024-02-28 om 09 34 19" src="https://github.com/Budibase/budibase/assets/157612513/1926ba6c-cd28-4f52-ad09-4991abe07a96">

I hope to get this solutions work. It will save me al lot of work and time. 
I have imported data into a table in which a column with data needs to be moved to my relationship column. In other words, with the existing data in two tables, the relationship columns in both tables must be filled. Can I use your example for this?

<img width="1385" alt="Scherm­afbeelding 2024-02-28 om 09 42 46" src="https://github.com/Budibase/budibase/assets/157612513/59c3c7d3-6863-4553-a530-978125099021">
